tests/unit/data/gomod-mocks/generate.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+
+git clone https://github.com/cachito-testing/gomod-pandemonium
```

Comment:
The unit tests are based around this repo. It exposes the one corner cases of our chosen module identification method - a [locally replaced module](https://github.com/cachito-testing/gomod-pandemonium/blob/a18784b5653c5ca26a2efa7d47143d0274bf5e70/go.mod#L35) which should be listed but isn't (because it's [excluded by build constraints](https://github.com/cachito-testing/gomod-pandemonium/blob/a18784b5653c5ca26a2efa7d47143d0274bf5e70/where-was-i-built/idk.go#L1)).

You can see that when vendoring, this module is reported (as expected), but when not vendoring, it isn't. That's the corner case identified in https://github.com/chmeliik/godeps#conclusion
```diff
❯ diff tests/unit/data/gomod-mocks/resolve_gomod_output.json tests/unit/data/gomod-mocks/resolve_gomod_output_vendored.json  --color
29a30,34
>       "name": "github.com/cachito-testing/gomod-pandemonium/weird",
>       "version": "./weird"
>     },
```
I can also add two integration test cases for this repo (with/without vendoring) if you think it's worth it, or conversely, I can base the unit tests around a less corner-case-y repo :smile:

Reply:
Hmm, that's a good question. Re-generating the data whenever we update Go definitely makes sense, but it's not something we've ever done before (in general, testing different Go versions has never been a thing, most int tests are still on Go 1.14 or something)

I'll add the Makefile command at least, and leave a comment in the "process for updating Go versions" JIRA